### PR TITLE
Add new_test/test_requires_unified_shared_memory_heap_map.F90

### PR DESCRIPTION
**Pull request by seyonglee**
NVHPC/21.11 passes the test.

### CODE REVIEW

File: tests/5.0/requires/test_requires_unified_shared_memory_heap_map.F90 [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    INTEGER, ALLOCATABLE:: anArray(:)
+    INTEGER, DIMENSION(N):: anArrayCopy
+
+    OMPVV_INFOMSG("Unified shared memory testing - Array on heap")
+
+    errors = 0
+
+    ALLOCATE(anArray(N))
+
+    OMPVV_ERROR_IF(.NOT. ALLOCATED(anArray), "Memory was not properly allocated")
```
Comment (spophale) [inline tests/5.0/requires/test_requires_unified_shared_memory_heap_map.F90:47]: The test should exit if the allocation failed.

**Comment by seyonglee** [comment]
[Update on August 16, 2022]
        - Inserted error message and exit code when the heap memory allocation fails.
        - NVHPC 22.5: C test failed but Fortran test passed.
        - GCC 12.1.1: C test passed but Fortran test failed.
        - XL 16.1.1-10: C test passed but ran on host. Fortran test failed.

File: tests/5.0/requires/test_requires_unified_shared_memory_heap_map.F90 (hunk 1 above)
Comment (seyonglee) [reply tests/5.0/requires/test_requires_unified_shared_memory_heap_map.F90]: Updated.

**Review by spophale — APPROVED** [review]

**Comment by spophale** [comment]
@fel-cab , @krishols will review.

**Review by krishols — APPROVED** [review]

**Review by fel-cab — APPROVED** [review]

**Pull request merged**